### PR TITLE
feat(container):  add resource container namespace

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

File: docs/data-sources/container_namespace.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# scaleway_container_namespace
+
+Gets information about a container namespace.
+
+## Example Usage
+
+```hcl
+// Get info by namespace name
+data "scaleway_container_namespace" "my_namespace" {
```
Comment (remyleone) [inline docs/data-sources/container_namespace.md:15]:
```suggestion
data "scaleway_container_namespace" "by_name" {
```

File: docs/data-sources/container_namespace.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  name = "my-namespace-name"
+}
+
+// Get info by namespace ID
+data "scaleway_container_namespace" "my_namespace" {
```
Comment (remyleone) [inline docs/data-sources/container_namespace.md:20]:
```suggestion
data "scaleway_container_namespace" "by_id" {
```

File: scaleway/resource_container_namespace.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	return &schema.Resource{
+		CreateContext: resourceScalewayContainerNamespaceCreate,
+		ReadContext:   resourceScalewayContainerNamespaceRead,
+		UpdateContext: resourceScalewayContainerNamespaceUpdate,
+		DeleteContext: resourceScalewayContainerNamespaceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultContainerTimeout),
```
Comment (remyleone) [inline scaleway/resource_container_namespace.go:23]: Should be defaultContainerNamespaceTimeout other resource related to container will have timeout

Comment (Monitob) [reply scaleway/resource_container_namespace.go]: Constant updated

File: scaleway/resource_container_namespace.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+func resourceScalewayContainerNamespaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := containerAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ns, err := api.CreateNamespace(&container.CreateNamespaceRequest{
+		Description:          expandStringPtr(d.Get("description").(string)),
+		EnvironmentVariables: expandMapStringStringPtr(d.Get("environment_variables")),
+		Name:                 expandOrGenerateString(d.Get("name").(string), "scw-container"),
```
Comment (remyleone) [inline scaleway/resource_container_namespace.go:75]:
```suggestion
		Name:                 expandOrGenerateString(d.Get("name").(string), "ns"),
```

Comment (remyleone) [reply scaleway/resource_container_namespace.go]: They have issues with too long names sometimes

**Review by remyleone — APPROVED** [review]

**Pull request merged**